docs/source/plugin/anatomy.rst

```diff
@@ -0,0 +1,345 @@
+===================
+Anatomy of a plugin
+===================
+
+A Sopel plugin consists of a Python module containing one or more
+``callable``\s. It may optionally also contain ``configure``, ``setup``, and
+``shutdown`` hooks.
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Defining rules
+==============
+
+The main goal of a Sopel plugin is to react to IRC messages. For that, Sopel
+uses a :term:`Rule system`: plugins define rules, Sopel loads them and triggers
+them when a message matches.
+
+Sopel identifies a callable as a rule when it has been decorated with any of
+these :mod:`sopel.plugin`'s decorators:
+
+* :term:`Generic rule`:: :func:`~sopel.plugin.rule`,
+  :func:`~sopel.plugin.find`, and :func:`~sopel.plugin.search`
+* :term:`Named rule`: :func:`~sopel.plugin.commands`, 
+  :func:`~sopel.plugin.action_commands`, and
+  :func:`~sopel.plugin.nickname_commands`
+* :term:`URL callback`: :func:`~sopel.plugin.url` and
+  :func:`~sopel.plugin.url_lazy`
+
+Additionnaly, Sopel identifies a callable as a generic rule when these
+decorator are used alone:
+
+* event based rule: :func:`~sopel.plugin.event`
+* intent/CTCP based rule: :func:`~sopel.plugin.intent`
+
+In that case, it will use a match-all regex (`r'.*'`)::
+
+   from sopel import plugin
+
+   @plugin.event('JOIN')
+   def on_join_2(bot, trigger):
+      pass
+
+   # this is equivalent to this:
+   @plugin.rule(r'.*')
+   @plugin.event('JOIN')
+   def on_join(bot, trigger):
+      pass
+
+Channel vs Private message
+--------------------------
+
+By default, :term:`rules <Rule>` can be triggered from a channel or a private
+message. It is possible to limit that to either one of these options:
+
+* channel only: :func:`sopel.plugin.require_chanmsg`
+* private message only: :func:`sopel.plugin.require_privmsg`
+
+Access right requirements
+-------------------------
+
+By default anyone can trigger a :term:`rule <Rule>`, and for some it might be
+better to limit who can trigger them. There are decorators for that:
+
+* :func:`sopel.plugin.require_account`: requires services/NickServ
+  authentication; works only of the server implement authentication properly
```

Comment:
```suggestion
  authentication; works only if the server implements modern IRC authentication
```

It's not "properly" per se. Just a newer standard that older networks usually don't support.

Reply:
I reused some part of the decorator's docstring for that.